src/Datadog.Trace/RuntimeMetrics/RuntimeMetricsWriter.cs

```diff
@@ -63,7 +63,7 @@ internal RuntimeMetricsWriter(IDogStatsd statsd, TimeSpan delay, Func<IDogStatsd
 
             try
             {
-                _listener = initializeListener(statsd, delay);
+                _listener = initializeListener?.Invoke(statsd, delay) ?? InitializeListener(statsd, delay);
```

Comment:
You're changing the behavior here. It means you will call `InitializeListener` if `initializeListener` returns a null value, which wasn't the case before

Reply:
oh yeah, that's right, reverting that.

Reply:
done in b3feead